### DISCUSSION TITLE
Please remove "Setting max connections" in postgresql container.

### DISCUSSION THREAD

**Discussion by Llewellynvdm** [nextcloud/all-in-one#3055] [Questions] [answered]
https://github.com/nextcloud/all-in-one/blob/b076d5ea1a0f16e57164a5f3c45326ed180f3b91/Containers/postgresql/start.sh#L153

Can you please either remove this, or add a `ENV` value with which to override this, thanks.

# You ask why?

This limitation seems to block that deployment of all-in-one from being used in production, or for more than three users, or for more that 1TB of files... what ever that means, the amount of connections hit this limit and the stalls the server.

When I remove this limitation, this stalling and freezing-up of the system does not happen, but the moment and update runs, and my "hack" gets removed... we are back at freezing mode.

So if this is a critical feature then please add a global ENV option for us to set this to a value of our choosing.

I have read many forum posts around this topic and I am not sure why this should even be a question in a OpenSource project with GNU AFFERO GENERAL PUBLIC LICENSE. As this very agreement allows the end-user all the freedom they need, and since this limit prevent some of that freedom, I would strongly recommend it be made adjustable. Please :)

**Comment by szaimen** [comment]
We have something better in mind: https://github.com/nextcloud/all-in-one/pull/3021

**Comment by szaimen** [comment]
You could already try this: https://github.com/nextcloud/all-in-one/issues/3030#issuecomment-1649968776 and report back if it works

**Reply by szaimen** [reply]
This is now released with v7.0.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by Llewellynvdm** [reply]
This still does not resolve the issue, my answer still remains the only solution that does not cause the system to freeze.

**Reply by szaimen** [reply]
So you already upgraded to 7.0.0 and the issue still appears?

How much RAM do you have and which optional addons did you enable from the AIO interface?

**Reply by Llewellynvdm** [reply]
16 GB RAM with the default add-ons.
![image](https://github.com/nextcloud/all-in-one/assets/5607939/7f1185ac-7101-49c6-a180-d2c1345fbd5d)

**Reply by szaimen** [reply]
How many users do you have?

**Reply by Llewellynvdm** [reply]
We have 12 users

**Reply by Llewellynvdm** [reply, marked as answer]
The issue seems to now be resolved, thank you! @szaimen  can mark this (reply) now as the correct answer.

**Comment by Llewellynvdm** [comment]
This is my little hack:

# How to fix the MAX_CONNECTIONS limit

This causes the system to freeze/hang when many activity is done on the system. The default seems way to low, (350+-) I bump it to (2500) but I think we can even take it higher.

### You will need access the server!

Open the master container:
```
docker exec -it --user root  nextcloud-aio-mastercontainer /bin/bash
```

Then edit this file:
```
vi php/containers.json
```

Look for the object of the database:
```
"container_name": "nextcloud-aio-database"
```

To edit:
```
press: [i]
```

Then add the following mount point (check the last volume /start.sh):
```
     "volumes": [                                  
        {                                           
          "source": "nextcloud_aio_database",       
          "destination": "/var/lib/postgresql/data",
          "writeable": true                         
        },                                          
        {                                           
          "source": "nextcloud_aio_database_dump",  
          "destination": "/mnt/data",               
          "writeable": true                         
        },                                          
        {                                           
          "source": "/home/ubuntu/db/start.sh",  
          "destination": "/start.sh",               
          "writeable": true                         
        }
     ]
```

When done editing the file:
```
press: [Esc]
```

Then to save changes:
```
press: [:w] and [enter]
```

To close vi editor:
```
press: [:q]
```

Check the file on the parent system:
```
/home/ubuntu/db/start.sh
```

That it is up to date with:
```
https://github.com/nextcloud/all-in-one/blob/main/Containers/postgresql/start.sh
```

But update the following line that says: (Setting max connections...):

Change it to the following:
```
echo "Setting max connections for MORE"
# MEMORY=$(awk '/MemTotal/ {printf "%d", $2/1024}' /proc/meminfo
MAX_CONNECTIONS=2500
```
You can make it more... if 2500 is still not high enough.

Once all that is done, restart the whole system, go to:
```
https://nextcloud.domain.com/settings/admin/overview
```

Click on [Open Nextcloud AIO Interface ↗](https://xxx.xxx.x.x:8080/containers) and **stop** all containers and **start** them all again.

If you can't access `https://nextcloud.domain.com/` then go directly to:
```
https://xxx.xxx.x.x:8080/containers
```

**Comment by Llewellynvdm** [comment]
I have been following the changes made to the system in the last two months closely and I am not sure why things are increasingly getting harder for us to override the defaults of the various containers. I realize there must be some good objective but one of the changes made... like this one: https://github.com/nextcloud/all-in-one/commit/581456b45069b7c8a8a22d7c092182bf38acd670 does not make sense to me. Why add this command: `chmod -R 777 /var/run/postgresql;` which any way is not allowed, and serious bad practice: `chmod: /var/run/postgresql: Operation not permitted`
![image](https://github.com/nextcloud/all-in-one/assets/5607939/5c7b1a57-f080-4bb1-8d6c-488b3300dbe4)

Yet this stops us from adding a simpler solution to the above issue. So hope you understand why it all seems weird to me. Previously it was easy to just change the `start.sh` directly but now its no longer possible, now we need this whole workaround, because of the read_only added to the `php/containers.json`

But like I said there must be so good objective here, its just messing with us little guys I suppose... do you mind explaining the objective, or point me to the road map?

**Comment by Llewellynvdm** [comment]
@szaimen this is disappointing. The issue remains unresolved, and your abuse of administrative rights is enough to alienate us from you.

**Reply by szaimen** [reply]
Hi, as I said this will be resolved by setting the dbpersistent value in nextcloud which you can either do manually now or wait for the next update which will do the same.
